haystack/tools/pipeline_tool.py

```diff
@@ -0,0 +1,240 @@
+
+    ## Usage Example:
+
+    ```python
+    from haystack import Document, Pipeline
+    from haystack.dataclasses import ChatMessage
+    from haystack.document_stores.in_memory import InMemoryDocumentStore
+    from haystack.components.embedders.sentence_transformers_text_embedder import SentenceTransformersTextEmbedder
+    from haystack.components.embedders.sentence_transformers_document_embedder import (
+        SentenceTransformersDocumentEmbedder
+    )
+    from haystack.components.generators.chat import OpenAIChatGenerator
+    from haystack.components.retrievers import InMemoryEmbeddingRetriever
+    from haystack.components.agents import Agent
+    from haystack.tools import PipelineTool
+
+    # Initialize a document store and add some documents
+    document_store = InMemoryDocumentStore()
+    document_embedder = SentenceTransformersDocumentEmbedder(model="sentence-transformers/all-MiniLM-L6-v2")
+    documents = [
+        Document(content="Nikola Tesla was a Serbian-American inventor and electrical engineer."),
+        Document(
+            content="He is best known for his contributions to the design of the modern alternating current (AC) "
+                    "electricity supply system."
+        ),
+    ]
+    document_embedder.warm_up()
+    docs_with_embeddings = document_embedder.run(documents=documents)["documents"]
+    document_store.write_documents(docs_with_embeddings)
+
+    # Build a simple retrieval pipeline
+    retrieval_pipeline = Pipeline()
+    retrieval_pipeline.add_component(
+        "embedder", SentenceTransformersTextEmbedder(model="sentence-transformers/all-MiniLM-L6-v2")
+    )
+    retrieval_pipeline.add_component("retriever", InMemoryEmbeddingRetriever(document_store=document_store))
+
+    retrieval_pipeline.connect("embedder.embedding", "retriever.query_embedding")
+
+    # Wrap the pipeline as a tool
+    retriever_tool = PipelineTool(
+        pipeline=retrieval_pipeline,
+        input_mapping={"query": ["embedder.text"]},
+        output_mapping={"retriever.documents": "documents"},
+        name="document_retriever",
+        description="Retrieve documents relevant to a query from the document store",
+    )
+
+    # Create an Agent with the tool
+    agent = Agent(
+        chat_generator=OpenAIChatGenerator(model="gpt-4.1-mini"),
+        tools=[retriever_tool]
+    )
+
+    # Let the Agent handle a query
+    result = agent.run([ChatMessage.from_user("Who was Nikola Tesla?")])
+
+    # Print result of the tool call
+    print("Tool Call Result:")
+    print(result["messages"][2].tool_call_result.result)
+    print("")
+
+    # Print answer
+    print("Answer:")
+    print(result["messages"][-1].text)
+    ```
+    """
+
+    def __init__(
+        self,
+        pipeline: Pipeline,
+        *,
+        name: str,
+        input_mapping: Optional[dict[str, list[str]]] = None,
+        output_mapping: Optional[dict[str, str]] = None,
+        description: Optional[str] = None,
```

Comment:
I'm considering making description also mandatory (like name) b/c otherwise the autogenerated description will just be the docstring of `SuperComponent` which doesn't seem helpful.

Reply:
okay I made it required